E2 UDF return type error

For data types such as vector and matrix where there're 3 types for each, UDFs doesn't take the type it's in into effect so running code like this...

```function matrix testing()
{   
    return matrix4()
}```
or

```function vector testing()
{   
    return vec4()
}```

Will return this error in e2...

```addons/wiremod/lua/entities/gmod_wire_expression2/base/compiler.lua:69: attempt to index a nil value```

```
> print(tps_pretty("ns"))
normal, string
> print(tps_pretty("xv4"))

[ERROR] addons/wiremod/lua/entities/gmod_wire_expression2/base/compiler.lua:69: attempt to index a nil value
  ...
```

--- a/lua/entities/gmod_wire_expression2/base/compiler.lua
+++ b/lua/entities/gmod_wire_expression2/base/compiler.lua
@@ -63,11 +63,11 @@ end
 
 function tps_pretty(tps)
 	if not tps or #tps == 0 then return "void" end
-
+	if type(tps) == "string" then tps = { tps } end
 	local ttt = {}
 	for i = 1, #tps do
-		ttt[i] = string.lower(wire_expression_types2[tps[i]][1])
-		if ttt[i] == "NORMAL" then ttt[i] = "number" end
+		local _, typenames = E2Lib.splitType(tps[i])
+		for j = 1, #typenames do table.insert(ttt, typenames[j]) end
 	end
 	return table.concat(ttt, ", ")
 end

--- a/lua/entities/gmod_wire_expression2/core/e2lib.lua
+++ b/lua/entities/gmod_wire_expression2/core/e2lib.lua
@@ -75,7 +75,7 @@ end
 
 -- -------------------------- signature generation -----------------------------
 
-local function maketype(typeid)
+function E2Lib.typeName(typeid)
 	if typeid == "" then return "void" end
 	if typeid == "n" then return "number" end
 
@@ -86,7 +86,7 @@ local function maketype(typeid)
 	return typename or "unknown"
 end
 
-local function splitargs(args)
+function E2Lib.splitType(args)
 	local ret = {}
 	local thistype
 	local i = 1
@@ -109,7 +109,7 @@ local function splitargs(args)
 				typeid = args:sub(i, i + 2)
 				i = i + 2
 			end
-			table.insert(ret, maketype(typeid))
+			table.insert(ret, E2Lib.typeName(typeid))
 		end
 		i = i + 1
 	end
@@ -121,7 +121,7 @@ function E2Lib.generate_signature(signature, rets, argnames)
 	local funcname, args = string.match(signature, "([^(]+)%(([^)]*)%)")
 	if not funcname then error("malformed signature") end
 
-	local thistype, args = splitargs(args)
+	local thistype, args = E2Lib.splitType(args)
 
 	if argnames then
 		for i = 1, #args do
@@ -131,7 +131,7 @@ function E2Lib.generate_signature(signature, rets, argnames)
 	local new_signature = string.format("%s(%s)", funcname, table.concat(args, ","))
 	if thistype then new_signature = thistype .. ":" .. new_signature end
 
-	return (not rets or rets == "") and (new_signature) or (maketype(rets) .. "=" .. new_signature)
+	return (not rets or rets == "") and (new_signature) or (E2Lib.typeName(rets) .. "=" .. new_signature)
 end
 
 -- ------------------------ various entity checkers ----------------------------

